server/plugins/schema-plugin-ob-mysql/src/main/java/com/oceanbase/odc/plugin/schema/obmysql/parser/OBMySQLGetDBTableByParser.java

```diff
@@ -300,32 +299,37 @@ private void fillSubPartitions(DBTablePartition partition, Partition partitionSt
                 .setPartitionsNum(
                         parsedSubPartitionOption.getTemplates() != null ? parsedSubPartitionOption.getTemplates().size()
                                 : null);
-        for (PartitionElement partitionElement : partitionStmt.getPartitionElements()) {
-            if (partitionElement.getSubPartitionElements() != null) {
+        for (int i = 0; i < partitionStmt.getPartitionElements().size(); i++) {
+            DBTablePartitionDefinition partitionDefinition = partition.getPartitionDefinitions().get(i);
+            if (partitionStmt.getPartitionElements().get(i).getSubPartitionElements() != null) {
                 // obtain DBTablePartitionDefinitions for non-templated subpartitions
-                for (int i = 0; i < partitionElement.getSubPartitionElements().size(); i++) {
+                for (int j = 0; j < partitionStmt.getPartitionElements().get(i).getSubPartitionElements().size(); j++) {
```

Comment:
this branch is too deep, suggest to cut in several smaller function to implement your logic.
by the way,  Array.get(index) query speed depends on it's arrayList or linkedList. Unless index is required, use iterator is more efficient. 
and all the time, npe defend process is not seen in this code

Reply:
1. 2 submethods are added to implement template and non-template logical processing
2.index is required
3. will add npe check